plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java

```diff
@@ -1305,6 +1305,49 @@ private void assertPartitionValuesParsedCondition(String tableName, int id, @Lan
                 .matches("VALUES " + id);
     }
 
+    @Test
+    public void testReadV2Checkpoint()
+            throws Exception
+    {
+        testReadV2Checkpoint("deltalake/v2_checkpoint_json");
+        testReadV2Checkpoint("deltalake/v2_checkpoint_parquet");
+        testReadV2Checkpoint("databricks133/v2_checkpoint_json");
+        testReadV2Checkpoint("databricks133/v2_checkpoint_parquet");
+    }
+
+    private void testReadV2Checkpoint(String resourceName)
+            throws Exception
+    {
+        String tableName = "test_v2_checkpoint_" + randomNameSuffix();
+        Path tableLocation = Files.createTempFile(tableName, null);
+        copyDirectoryContents(new File(Resources.getResource(resourceName).toURI()).toPath(), tableLocation);
+
+        assertUpdate("CALL system.register_table('%s', '%s', '%s')".formatted(getSession().getSchema().orElseThrow(), tableName, tableLocation.toUri()));
+        assertThat(query("DESCRIBE " + tableName))
+                .projected("Column", "Type")
+                .skippingTypesCheck()
+                .matches("VALUES ('a', 'integer'), ('b', 'integer')");
+        assertQuery("SELECT * FROM " + tableName, "VALUES (1, 2)");
```

Comment:
Any way to test that checkpoint is actually read? Maybe deleting other transaction log files from resources?

Reply:
I think verifying the existence of _last_checkpoint is sufficient. Added another assertion.